### ISSUE TITLE
Building of tests fails with gcc7 due to Werror

### ISSUE THREAD

**Issue by heirecka** [msgpack/msgpack-c#598]
With 2.1.2:

```
[ 48%] Building CXX object test/CMakeFiles/msgpack_stream.dir/msgpack_stream.cpp.o
cd /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/build/test && /usr/bin/x86_64-pc-linux-gnu-c++   -I/var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/include -I/var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/build/include  -DMSGPACK_DEFAULT_API_VERSION=2 -DMSGPACK_USE_BOOST -std=c++11 -march=native -O2 -pipe   -Wall -Wextra -Werror -g -O3  -o CMakeFiles/msgpack_stream.dir/msgpack_stream.cpp.o -c /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/test/msgpack_stream.cpp
In file included from /usr/x86_64-pc-linux-gnu/include/gtest/gtest.h:1874:0,
                 from /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/test/msgpack_c.cpp:7:
/var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/test/msgpack_c.cpp: In member function 'virtual void MSGPACKC_simple_buffer_v4raw_fix_l_Test::TestBody()':
/var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/test/msgpack_c.cpp:978:24: error: argument 1 null where non-null expected [-Werror=nonnull]
     EXPECT_EQ(0, memcmp(str, obj.via.str.ptr, str_size));
                  ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/include/msgpack/pack_define.h:15:0,
                 from /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/include/msgpack/pack.h:13,
                 from /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/include/msgpack.h:19,
                 from /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/test/msgpack_c.cpp:1:
/usr/x86_64-pc-linux-gnu/include/string.h:66:12: note: in a call to function 'int memcmp(const void*, const void*, size_t)' declared here
 extern int memcmp (const void *__s1, const void *__s2, size_t __n)
            ^~~~~~
In file included from /usr/x86_64-pc-linux-gnu/include/gtest/gtest.h:1874:0,
                 from /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/test/msgpack_c.cpp:7:
/var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/test/msgpack_c.cpp: In member function 'virtual void MSGPACKC_simple_buffer_str_fix_l_Test::TestBody()':
/var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/test/msgpack_c.cpp:779:24: error: argument 1 null where non-null expected [-Werror=nonnull]
     EXPECT_EQ(0, memcmp(str, obj.via.str.ptr, str_size));
                  ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/include/msgpack/pack_define.h:15:0,
                 from /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/include/msgpack/pack.h:13,
                 from /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/include/msgpack.h:19,
                 from /var/tmp/paludis/build/dev-libs-msgpack-2.1.2/work/msgpack-c-cpp-2.1.2/test/msgpack_c.cpp:1:
/usr/x86_64-pc-linux-gnu/include/string.h:66:12: note: in a call to function 'int memcmp(const void*, const void*, size_t)' declared here
 extern int memcmp (const void *__s1, const void *__s2, size_t __n)
```

**Comment by redboltz** [comment]
@heirecka , thank you for reporting the issue.

I will fix the issue.

**Comment by redboltz** [comment]
Fixed by #599 (merged). Could you check it? If the problem is not solved, please reopen the issue.

**Comment by heirecka** [comment]
Thanks! a7a78bd does indeed fix the issue.
